### PR TITLE
[ML][Pipelines] Support queue settings in pipeline job

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Previous PR  #29159 (Command) and #29204 (AutoML & Sweep) introduce `queue_settings`, this PR targets to support this in pipeline job.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/pipeline_job/unittests/test_pipeline_job_entity.py [hunk 1]
```diff
@@ -2034,3 +2034,10 @@ def empty_value_pipeline(integer: int, boolean: bool, number: float, str_param:
         rest_obj = pipeline._to_rest_object()
         expect_resource = {"instance_count": "${{parent.inputs.integer}}", "shm_size": "${{parent.inputs.shm_size}}"}
         assert rest_obj.properties.jobs["component"]["resources"] == expect_resource
+
+    def test_pipeline_job_with_queue_settings(self) -> None:
+        yaml_path = "./tests/test_configs/pipeline_jobs/serverless_compute/pipeline_with_queue_settings.yml"
+        pipeline_job = load_job(yaml_path)
+        rest_obj = pipeline_job._to_rest_object()
+        expected_queue_settings = {"job_tier": "Standard", "priority": 2}
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/pipeline_job/unittests/test_pipeline_job_entity.py:2042]: add an e2e test? we can skip if that does not work yet.

job_tier could be tested with serverless compute.
"job_tier": & "priority" should be tested with singularity case

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/pipeline_job/unittests/test_pipeline_job_entity.py:2042]: Sure, a submit & cancel E2E test should pass currently. Let me update this for `job_tier`, as we target serverless compute in this PR.

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/pipeline_job/unittests/test_pipeline_job_entity.py]: Updated this UT & added one E2E test, may take a look.

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 2]
```diff
@@ -554,7 +554,7 @@ def _to_job(self) -> CommandJob:
 
     @classmethod
     def _picked_fields_from_dict_to_rest_object(cls) -> List[str]:
-        return ["resources", "distribution", "limits", "environment_variables"]
+        return ["resources", "distribution", "limits", "environment_variables", "queue_settings"]
```
Comment (sweana) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:557]: Do you need similar change for AutoML?

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:557]: No, and I have one test for AutoML in Pipelines, seems it works well.

**Pull request merged**